reporting/src/main/java/org/wso2/testgrid/reporting/EmailReportGenerator.java

```diff
@@ -0,0 +1,205 @@
+                    testPlan.getDeploymentPattern().getProduct().getName() + "having infra-combination as " +
+                    testPlan.getInfraParameters(), e);
+        }
+
+        if (Files.exists(filePath)) {
+            try (BufferedReader bufferedReader = Files.newBufferedReader(filePath)) {
+                String currentLine;
+                int lineCount = 0;
+                while ((currentLine = bufferedReader.readLine()) != null) {
+                    stringBuilder.append(currentLine).append(HTML_LINE_SEPARATOR);
+                    if (++lineCount == 100) {
+                        stringBuilder.append(".........").append(HTML_LINE_SEPARATOR).append(".........")
+                                .append(HTML_LINE_SEPARATOR).append("(view complete log in detailed results..)");
+                        break;
+                    }
+                }
+                return stringBuilder.toString();
+            } catch (IOException e) {
+                throw new ReportingException("Error occurred while reading integration-test log file to get" +
+                        " integration test results for test-plan with id " + testPlan.getId() + "of the product " +
+                        testPlan.getDeploymentPattern().getProduct().getName() + "having infra-combination as " +
+                        testPlan.getInfraParameters(), e);
+            }
+        } else {
+            if (testPlan.getStatus().equals(Status.SUCCESS) || testPlan.getStatus().equals(Status.FAIL)) {
+                logger.error("Integration-test log file does not exists for test-plan with id "
+                        + testPlan.getId() + "of the product " +
+                        testPlan.getDeploymentPattern().getProduct().getName() + "having infra-combination as " +
+                        testPlan.getInfraParameters());
+                return "Integration test-log is missing. Please contact TestGrid administrator.";
+            } else {
+                return "Test status is " + testPlan.getStatus() + ". Hence can not display the log.";
+            }
+        }
+    }
+
+    /**
+     * Returns the current status of the product.
+     * @param product product
+     * @return current status of the product
+     */
+    public Status getProductStatus(Product product) {
+        TestPlanUOW testPlanUOW = new TestPlanUOW();
+        return testPlanUOW.getCurrentStatus(product);
+    }
+
+    /**
+     * Returns the Git build information of the latest build of the product.
+     * This will consist of git location and git revision used to build the distribution.
+     * @param product product
+     * @return Git build information
+     */
+    public String getGitBuildDetails(Product product) {
+        StringBuilder stringBuilder = new StringBuilder();
+        TestPlanUOW testPlanUOW = new TestPlanUOW();
+        List<TestPlan> testPlans = testPlanUOW.getLatestTestPlans(product);
+        TestPlan testPlan = testPlans.get(0);
```

Comment:
It would be good to add a comment here explaining the logic here. We are doing this because the git build details will be similar across all test-plans. But, for a fresh eye, it may not be.